src/hooks/useInstances.tsx

```diff
@@ -1,69 +1,159 @@
+import { AxiosRequestConfig, Method as AxiosMethod } from 'axios'
+import useAxios, { Options as AxiosHooksOptions } from 'axios-hooks'
 import { ServerConfigContainer } from 'containers/ConfigContainer'
 import { useMyAxios } from 'hooks/useMyAxios'
-import { Instance } from 'types/backend-types'
+import { Instance, System } from 'types/backend-types'
 
-export const useInstances = () => {
-  const { authEnabled } = ServerConfigContainer.useContainer()
-  const { axiosInstance } = useMyAxios()
+const axiosOptions: AxiosHooksOptions = {
+  manual: true,
+  useCache: false,
+  ssr: false,
+  autoCancel: false,
+}
 
-  const stopInstance = (id: string) => {
-    axiosInstance.patch<Instance>(
-      `/api/v1/instances/${id}`,
-      {
-        operation: 'stop',
-      },
-      {
-        withCredentials: authEnabled,
-      },
-    )
+const getConfig = (id: string, method: AxiosMethod, authEnabled: boolean) => {
+  const config: AxiosRequestConfig<Instance> = {
+    url: `/api/v1/instances/${id}`,
+    method: method,
+    withCredentials: authEnabled,
   }
+  return config
+}
 
-  const startInstance = (id: string) => {
-    axiosInstance.patch<Instance>(
-      `/api/v1/instances/${id}`,
-      {
-        operation: 'start',
-      },
-      {
-        withCredentials: authEnabled,
-      },
-    )
+const useFetchStatus = () => {
+  const { authEnabled } = ServerConfigContainer.useContainer()
+
+  const [, execute] = useAxios({}, axiosOptions)
+
+  /**
+   * Fetches status of an instance from the backend after a timeout
+   *
+   * @param id instance id
+   * @returns
+   */
+  const fetchStatus = (id: string) => {
+    const config = getConfig(id, 'get', authEnabled)
+
+    const promise = new Promise<string>((resolve, reject) => {
+      setTimeout(async () => {
+        const response = execute(config).then(async (resp) => {
+          return resp.data.status
+        })
+        resolve(response)
+      }, 2000)
+    })
+
+    return promise
   }
 
-  const getInstance = (id: string) => {
-    axiosInstance.get<Instance>(`/api/v1/instances/${id}`, {
-      withCredentials: authEnabled,
+  return fetchStatus
+}
+
+const useToggleInstance = () => {
```

Comment:
This breakout into several different hooks was before I really got the pattern down. This works fine but there's a fair amount of duplication that can potentially be removed with yet another refactor to the pattern that's shown elsewhere.

Reply:
Since I had a little time, I decided to just refactor this now.